
From: Roger Luethi <rl@hellgate.ch>

If you win the race with a starting process, you can read its environment.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/fs/proc/base.c |    2 ++
 1 files changed, 2 insertions(+)

diff -puN fs/proc/base.c~proc_pid_cmdline-race-fix fs/proc/base.c
--- 25/fs/proc/base.c~proc_pid_cmdline-race-fix	2004-08-05 11:28:21.915442360 -0700
+++ 25-akpm/fs/proc/base.c	2004-08-05 11:28:21.919441752 -0700
@@ -340,6 +340,8 @@ static int proc_pid_cmdline(struct task_
 	struct mm_struct *mm = get_task_mm(task);
 	if (!mm)
 		goto out;
+	if (!mm->arg_end)
+		goto out;	/* Shh! No looking before we're done */
 
  	len = mm->arg_end - mm->arg_start;
  
_
